src/Cli/AzureCliLogin.ts

```diff
@@ -108,7 +110,14 @@ export class AzureCliLogin {
     }
 
     async loginWithUserAssignedIdentity(args: string[]) {
-        args.push("--username", this.loginConfig.servicePrincipalId);
+        const azcliMinorVersion = parseInt(this.azVersion.split('.')[1], 10);

```

Comment:
What if `parseInt` throws an exception? Will it fail the login?

Reply:
Not sure how `parseInt` could throw an exception on `azcliMinorVersion`. The Azure CLI version always follows the `xx.xx.xx` format.

Reply:
The second version section is always an integer.